lib/config.js

```diff
@@ -86,6 +87,12 @@ function loadConfig(filePath) {
             config = require(filePath);
         }
 
+        if (config.rules) {
```

Comment:
Since configs are just simple objects full of name/options pairs, options have to be validated when location information is still available. My first approach was to validate exactly once in `verify` when the final configuration had been determined, but we wanted to be able to print the source of any invalid options, so I moved validation out into the locations you listed.

I suppose the alternative approach would be to attach source location metadata to config objects, but that seemed like overkill.